includes/class-duplicate-payment-prevention-service.php

```diff
@@ -115,7 +115,28 @@ public function check_payment_intent_attached_to_order_succeeded( WC_Order $orde
 		if ( Intent_Status::SUCCEEDED === $intent_status ) {
 			$this->remove_session_processing_order( $order->get_id() );
 		}
-		$this->order_service->update_order_status_from_intent( $order, $intent );
+
+		// Check if the order amount matches the charged amount.
+		$order_total_in_cents = (int) ( $order->get_total() * 100 );
+		$charge               = $intent->get_charge();
+		$charged_amount       = $charge ? $charge->get_amount() : 0;
```

Comment:
You should be able to use the intent directly.
```suggestion
		$charged_amount       = $intent->get_amount()
```

Reply:
addressed: f117d7f4dc0babef6c7db0cf95dc5d9c3b806775